### PR TITLE
Enable AuthType auto selection when linking a HDInsight cluster

### PR DESCRIPTION
**Pull request by wezhang**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java [hunk 1]
```diff
@@ -313,6 +307,31 @@ protected void createUIComponents() {
...
+
+                if (StringUtils.isBlank(clusterNameOrUrl)) {
+                    return;
+                }
+
+                try {
+                    AuthType authType = SparkBatchSubmission.getInstance().probeAuthType(
+                            getClusterConnectionUrl(clusterNameOrUrl.trim()));
+
+                    authComboBox.setSelectedItem(authType);
```
Comment (t-rufang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java]: Could you do some tests to determine whether `probeAuthType` is a time consuming method and cause UI stuck?

Comment (t-rufang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java]: What if the authType is `AuthType.NotSupported`

Comment (wezhang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java]: In my test, nothing will happen if the type isn't matched.

Comment (wezhang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java]: About 500ms, you can feel that. 😄

**Pull request merged**